### PR TITLE
[mlir][test] Fix filecheck annotation typos [2/n]

### PR DESCRIPTION
**Pull request by klensy**
Few more fixes
previous: https://github.com/llvm/llvm-project/pull/92897 pr
Issues from https://github.com/llvm/llvm-project/issues/93154 unfixed.

### CODE REVIEW

**Comment by banach-space** [comment]
LG, thanks!

If this is patch K (e.g. 2 or 3) in a series of N patches, could you add a link to the previous patches in the summary? Also, could you and add sth like `(patch K/N)` to the title?  This way it will be very obvious that this is a part of a multi-PR refactor. Otherwise that's not that clear.

**Comment by klensy** [comment]
Added link to previous PR.
I didn't intended this be series of PRs, but just found new pattern of typos which revealed new errors.

**Comment by banach-space** [comment]
> Added link to previous PR.

Thanks!

> I didn't intended this be series of PRs, but just found new pattern of typos which revealed new errors.

This happens to me quite regularly :) Still, IMHO it's good to add a note like `(patch 2/N)` to indicate that this is 2nd patch in a series of similar patches (makes the context much clear). Note that there's no need to specify `N` if you don't know how many similar patches you'll end up sending (this is fairly common).

**Comment by klensy** [comment]
> I see that there are some CI failures - these need to be addressed before landing :)

I need help resolving these CI errors, as not all of them trivial.

**Comment by MacDue** [comment]
Fix for the arm streaming attributes:  
```diff
diff --git a/mlir/test/Target/LLVMIR/llvmir.mlir b/mlir/test/Target/LLVMIR/llvmir.mlir
index 97f37939551d..c83641a57537 100644
--- a/mlir/test/Target/LLVMIR/llvmir.mlir
+++ b/mlir/test/Target/LLVMIR/llvmir.mlir
@@ -2331,39 +2331,47 @@ llvm.func @streaming_compatible_func() attributes {arm_streaming_compatible} {
 // -----
 
 // CHECK-LABEL: @new_za_func
-// CHECK: #[[ATTR:[0-9]*]]
+// CHECK-SAME: #[[ATTR:[0-9]*]]
 llvm.func @new_za_func() attributes {arm_new_za} {
   llvm.return
 }
-// CHECK #[[ATTR]] = { "aarch64_new_za" }
+// CHECK: #[[ATTR]] = { "aarch64_new_za" }
+
+// -----
 
 // CHECK-LABEL: @in_za_func
-// CHECK: #[[ATTR:[0-9]*]]
+// CHECK-SAME: #[[ATTR:[0-9]*]]
 llvm.func @in_za_func() attributes {arm_in_za } {
   llvm.return
 }
-// CHECK #[[ATTR]] = { "aarch64_in_za" }
+// CHECK: #[[ATTR]] = { "aarch64_in_za" }
+
+// -----
 
 // CHECK-LABEL: @out_za_func
-// CHECK: #[[ATTR:[0-9]*]]
+// CHECK-SAME: #[[ATTR:[0-9]*]]
 llvm.func @out_za_func() attributes {arm_out_za } {
   llvm.return
 }
-// CHECK #[[ATTR]] = { "aarch64_out_za" }
+// CHECK: #[[ATTR]] = { "aarch64_out_za" }
+
+// -----
 
 // CHECK-LABEL: @inout_za_func
-// CHECK: #[[ATTR:[0-9]*]]
+// CHECK-SAME: #[[ATTR:[0-9]*]]
 llvm.func @inout_za_func() attributes {arm_inout_za } {
   llvm.return
 }
-// CHECK #[[ATTR]] = { "aarch64_inout_za" }
+// CHECK: #[[ATTR]] = { "aarch64_inout_za" }
+
+// -----
 
 // CHECK-LABEL: @preserves_za_func
-// CHECK: #[[ATTR:[0-9]*]]
+// CHECK-SAME: #[[ATTR:[0-9]*]]
 llvm.func @preserves_za_func() attributes {arm_preserves_za} {
   llvm.return
 }
-// CHECK #[[ATTR]] = { "aarch64_preserves_za" }
+// CHECK: #[[ATTR]] = { "aarch64_preserves_za" }
 
 // -----
```

**Comment by klensy** [comment]
> Fix for the arm streaming attributes:

done

**Comment by MacDue** [comment]
Here's a fix for `attributes.py` too:

```diff
diff --git a/mlir/test/python/ir/attributes.py b/mlir/test/python/ir/attributes.py
index dbd6bad05e01..6c1ac88f3226 100644
--- a/mlir/test/python/ir/attributes.py
+++ b/mlir/test/python/ir/attributes.py
@@ -514,7 +514,7 @@ def testDictAttr():
 
         a = DictAttr.get(dict_attr)
 
-        # CHECK attr: {integerattr = 42 : i32, stringattr = "string"}
+        # CHECK: attr: {integerattr = 42 : i32, stringattr = "string"}
         print("attr:", a)
 
         assert len(a) == 2
@@ -546,7 +546,7 @@ def testDictAttr():
         else:
             assert False, "expected IndexError on accessing an out-of-bounds attribute"
 
-        # CHECK "empty: {}"
+        # CHECK: empty: {}
         print("empty: ", DictAttr.get())
```

**Comment by klensy** [comment]
```
> -        # CHECK "empty: {}"
> +        # CHECK: empty: {}
```
ahh, quotes.

**Review by MacDue — APPROVED** [review]
Arm related changes look good :+1:

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

**Review by Dinistro — APPROVED** [review]
LLVM dialect related changes LGTM!
Thank you very much for pushing this cleanup effort 🙂

**Review by grypp — APPROVED** [review]

**Comment by aartbik** [comment]
sparse changes LGTM, thanks for fixing!

**Comment by MacDue** [comment]
Can we land this soon (and keep an eye on the buildbots for any extra failures)?

**Comment by Dinistro** [comment]
I'm pro landing this

**Comment by klensy** [comment]
NB: there still exist #93154 with few more wrong tests.

**Review by sjarus — APPROVED** [review]

File: mlir/test/IR/parser.mlir [hunk 1]
```diff
@@ -342,40 +342,40 @@ func.func @loop_bounds(%N : index) {
 
 // CHECK-LABEL: func @ifinst(%{{.*}}: index) {
 func.func @ifinst(%N: index) {
-  %c = arith.constant 200 : index // CHECK   %{{.*}} = arith.constant 200
-  affine.for %i = 1 to 10 {           // CHECK   affine.for %{{.*}} = 1 to 10 {
-    affine.if #set0(%i)[%N, %c] {     // CHECK     affine.if #set0(%{{.*}})[%{{.*}}, %{{.*}}] {
+  %c = arith.constant 200 : index // CHECK:  %{{.*}} = arith.constant 200
+  affine.for %i = 1 to 10 {           // CHECK:  affine.for %{{.*}} = 1 to 10 {
+    affine.if #set0(%i)[%N, %c] {     // CHECK:    affine.if #set(%{{.*}})[%{{.*}}, %{{.*}}] {
```
Comment (ftynse) [inline mlir/test/IR/parser.mlir:347]: We shouldn't hardcode transient names after `#`, same as variable and block names.

**Review by ftynse — APPROVED** [review]

**Comment by ftynse** [comment]
I'm going to just land this. Revert of fix-forward if something breaks.

File: mlir/test/IR/parser.mlir (hunk 1 above)
Comment (klensy) [reply mlir/test/IR/parser.mlir:347]: Yes, this looks weird, but there won't any better suggestions, so i've done what i've done.

**Pull request merged**